### PR TITLE
Update various tests to not use Swift 3 mode

### PR DESCRIPTION
**Pull request by slavapestov**

### CODE REVIEW

**Comment by slavapestov** [comment]
@swift-ci Please smoke test

File: stdlib/public/SDK/Foundation/IndexPath.swift [hunk 1]
```diff
@@ -265,7 +265,6 @@ public struct IndexPath : ReferenceConvertible, Equatable, Hashable, MutableColl
                     default:
                         fatalError("Range \(range) is out of bounds of count 1")
                     }
-                    return self
```
Comment (slavapestov) [inline stdlib/public/SDK/Foundation/IndexPath.swift]: @Catfish-Man Is this change OK?

File: test/Compatibility/MixAndMatch/Inputs/witness_change_swift4_leaf.swift [hunk 2]
```diff
@@ -1,70 +1,70 @@
 
-// Swift 3 sees the ObjC class NSRuncibleSpoon as the class, and uses methods
+// Swift 4 sees the ObjC class NSRuncibleSpoon as the class, and uses methods
```
Comment (jrose-apple) [inline test/Compatibility/MixAndMatch/Inputs/witness_change_swift4_leaf.swift]: Oops, thanks. I thought I got all of these.

**Review by jrose-apple — COMMENTED** [review]
👍

**Pull request merged**